smart-contract/contracts/ElectionFactory.sol

```diff
@@ -54,11 +54,11 @@ contract ElectionFactory {
     }
 
     /// @dev Deploys a new election smart contract and stores the details.
-    function createElection(string memory _position, string[] memory _contestants) external onlyOwner {
+    function createElection(string memory _position, string[] memory _contestants, address[] memory _contestantsAddress) external onlyOwner returns(address) {
```

Comment:
No need to return the address you can pick the address up from the event emitted